public/js/analytics.js

```diff
@@ -23,7 +23,16 @@ $(function () {
   $('#nav-analytics').addClass('active');
 
   // Loads the Student statistics by default
-  displayStudentStatistics(null);
+  if ($('#class-analytics-card').length) {
```

Comment:
what property does this check for? this seems like more of a hack that we should not be doing

Reply:
it checks if the student has access to class analytics, aka if the student is admin

Reply:
seems like a hack, please find a better way

Reply:
yes I can send an extra ajax request just to get the permission and to make you happy, would you like me to do so?

Reply:
there are a lot other ways to check that, navbar options for questions and feedback, fixing the pug to handle different boards, change the js files to split the boards, I just picked a simple if statement

Reply:
if we had time to implement the me object we wouldnt have ran into this issue, and a lot other issue

Reply:
all the options above are hacky solutions, your comment is not helping

Reply:
an `ajax` request to check the permission would be the correct thing to do here. the reason why is that this is a hack that we are relying on the default behavior of these classes, which is potential to change.